validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java

```diff
@@ -104,6 +104,15 @@ public void onNewHeadStateEvent(HeadStateEvent headStateEvent) {
       this.eventBus.post(attestation);
     }
 
+    List<Attestation> blockAttestations = headBlock.getBody().getAttestations();
+    synchronized (this.attestationsQueue) {
```

Comment:
What does this synchronized (this.attestationsQueue) do here?

Reply:
i don;t want attestation being added to the queue while i am looking through the list for potential dupes. which, btw, never finds any.   I feel like we need to remove already processed attestations from that queue as need blocks come in with attestations